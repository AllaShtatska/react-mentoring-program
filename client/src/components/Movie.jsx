import '../static/movie.css';
import '../static/image.css';
import '../static/inline.css';
import '../static/margin-left.css';

import React from 'react';

const Movie=({movie, onSelectMovie}) => (
    <div className='movie' onClick={()=>onSelectMovie(movie)}>
    <div className='image'><img src={movie.poster_path}/></div>
    <div className="inline">{movie.title}</div>
    <div className="inline margin-left">{movie.release_date}</div>
    <div>{movie.genres}</div>
    </div>
);

export default Movie;