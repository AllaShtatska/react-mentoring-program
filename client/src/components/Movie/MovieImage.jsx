import '../../static/image.css';

import React from 'react';

const MovieImage=({movie}) => (
    <div className='image'><img src={movie.poster_path}/></div>
);

export default MovieImage;