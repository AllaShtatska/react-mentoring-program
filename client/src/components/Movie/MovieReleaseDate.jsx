import '../../static/inline';
import '../../static/margin-left';

import React from 'react';

const MovieReleaseDate=({movie}) => (
    <div className="inline margin-left">{movie.release_date}</div>
);

export default MovieReleaseDate;