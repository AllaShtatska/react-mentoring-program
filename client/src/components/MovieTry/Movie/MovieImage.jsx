import '../../../static/image';

import React from 'react';

const MovieImage=({movie}) => (
    <div className='image'><img src={movie.poster_path}/></div>
);

export default MovieImage;