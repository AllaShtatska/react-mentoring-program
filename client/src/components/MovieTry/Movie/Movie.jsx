import '../../../static/movie';

import React from 'react';
import MovieImage from './MovieImage';
import MovieTitle from './MovieTitle';
import MovieReleaseDate from './MovieReleaseDate';
import MovieGenre from './MovieGenre'

const Movie=({movie, onSelectMovie}) => (
    <div className='movie' onClick={()=>onSelectMovie(movie)}><MovieImage movie={movie}/><MovieTitle movie={movie}/><MovieReleaseDate movie={movie}/><MovieGenre movie={movie}/></div>
);

export default Movie;