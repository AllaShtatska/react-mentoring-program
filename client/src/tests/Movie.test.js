import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Movie from '../components/Movie';

const movie = {
    id: "1",
    title: "Kill Bill: Vol1",
    genre: ["Actions&Advanture"],
    release_date: "October 10, 2003",
    vote_average: 10,
    poster_path: '../images/Kill_Bill_Volume_1.png',
    runtime: 111,
    overview: "Kill Bill: Vol1 overview"
};


it ('renders correctly', ()=>{
    const component = renderer.create(<Movie key={movie.id} movie={movie} onSelectMovie={function(){}}/>);

    var jsonComponent = component.toJSON();
    expect(jsonComponent).toMatchSnapshot();
});

it ('should trigger select movie on click', ()=>{
    const selectMovie = jest.fn();
    const component = shallow(<Movie key={movie.id} movie={movie} onSelectMovie={selectMovie}/>);

    component.find('.movie').simulate('click');

    expect(selectMovie).toHaveBeenCalled();
    expect(selectMovie).toBeCalledWith(movie);
});