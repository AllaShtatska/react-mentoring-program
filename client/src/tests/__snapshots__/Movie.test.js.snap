// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="movie"
  onClick={[Function]}
>
  <div
    className="image"
  >
    <img
      src="../images/Kill_Bill_Volume_1.png"
    />
  </div>
  <div
    className="inline"
  >
    Kill Bill: Vol1
  </div>
  <div
    className="inline margin-left"
  >
    October 10, 2003
  </div>
  <div>
    Actions&Advanture
  </div>
</div>
`;
